### ISSUE TITLE
The CSV file that is generated when we export the list of packages processed by an ingest job is baddly flormed

### ISSUE THREAD

**Issue by jmaferreira** [keeps/roda#633]
Look at the attachments. There is information there that shouldn't be on the CSV.

![screen shot 2016-10-20 at 00 29 09](https://cloud.githubusercontent.com/assets/3181302/19541192/6aa90782-965c-11e6-9ad4-f6496672f9d4.png)
![screen shot 2016-10-20 at 00 28 32](https://cloud.githubusercontent.com/assets/3181302/19541198/6ef4639a-965c-11e6-944e-864af73ce33b.png)
[Lista de processos.csv.zip](https://github.com/keeps/roda/files/540480/Lista.de.processos.csv.zip)

**Comment by rui-castro** [comment]
@jmaferreira "There is information there that shouldn't be on the CSV." What columns should be removed from the CSV? 
I would say "pluginDetails, htmlPluginDetails, reports".

Bellow is the data in json.

```json
{
  "offset": 0,
  "limit": 1,
  "totalCount": 1,
  "results": [
    {
      "id": "daa0133c-3149-45f3-ad1c-7acd921ab5e1-ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
      "jobId": "daa0133c-3149-45f3-ad1c-7acd921ab5e1",
      "sourceObjectId": "985d2729-b008-367e-8cb9-5917ee62948a",
      "sourceObjectClass": "org.roda.core.data.v2.ip.TransferredResource",
      "sourceObjectOriginalId": "",
      "sourceObjectOriginalName": "IMG_20160928_151524.jpg",
      "outcomeObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
      "outcomeObjectClass": "org.roda.core.data.v2.ip.AIP",
      "outcomeObjectState": "ACTIVE",
      "title": "Uploaded file/folder",
      "dateCreated": 1476977329181,
      "dateUpdated": 1476977346886,
      "completionPercentage": 100,
      "stepsCompleted": 7,
      "totalSteps": 7,
      "plugin": "org.roda.core.plugins.plugins.ingest.AutoAcceptSIPPlugin",
      "pluginName": "Auto accept",
      "pluginVersion": "1.0",
      "pluginState": "SUCCESS",
      "pluginDetails": "\n----------- SCAN SUMMARY -----------\nKnown viruses: 4961442\nEngine version: 0.99.2\nScanned directories: 7\nScanned files: 6\nInfected files: 0\nData scanned: 0.41 MB\nData read: 0.41 MB (ratio 1.01:1)\nTime: 14.341 sec (0 m 14 s)\nDone with checking producer authorization for AIP ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
      "htmlPluginDetails": false,
      "reports": [
        {
          "id": "daa0133c-3149-45f3-ad1c-7acd921ab5e1-ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "jobId": "daa0133c-3149-45f3-ad1c-7acd921ab5e1",
          "sourceObjectId": "985d2729-b008-367e-8cb9-5917ee62948a",
          "sourceObjectClass": "org.roda.core.data.v2.ip.TransferredResource",
          "sourceObjectOriginalId": "",
          "sourceObjectOriginalName": "IMG_20160928_151524.jpg",
          "outcomeObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "outcomeObjectClass": "org.roda.core.data.v2.ip.AIP",
          "outcomeObjectState": "INGEST_PROCESSING",
          "title": "Uploaded file/folder",
          "dateCreated": 1476977329181,
          "dateUpdated": 1476977331053,
          "completionPercentage": 0,
          "stepsCompleted": 0,
          "totalSteps": 7,
          "plugin": "org.roda.core.plugins.plugins.ingest.TransferredResourceToAIPPlugin",
          "pluginName": "Uploaded file/folder",
          "pluginVersion": "1.0",
          "pluginState": "SUCCESS",
          "pluginDetails": "",
          "htmlPluginDetails": false,
          "reports": []
        },
        {
          "id": "daa0133c-3149-45f3-ad1c-7acd921ab5e1-ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "jobId": "daa0133c-3149-45f3-ad1c-7acd921ab5e1",
          "sourceObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "sourceObjectClass": "org.roda.core.data.v2.ip.AIP",
          "sourceObjectOriginalId": "",
          "sourceObjectOriginalName": "",
          "outcomeObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "outcomeObjectClass": "org.roda.core.data.v2.ip.AIP",
          "outcomeObjectState": "INGEST_PROCESSING",
          "title": "AIP Virus check",
          "dateCreated": 1476977331174,
          "dateUpdated": 1476977345616,
          "completionPercentage": 0,
          "stepsCompleted": 0,
          "totalSteps": 7,
          "plugin": "org.roda.core.plugins.plugins.antivirus.AntivirusPlugin",
          "pluginName": "AIP Virus check",
          "pluginVersion": "ClamAV 0.99.2/22399/Thu Oct 20 14:55:43 2016",
          "pluginState": "SUCCESS",
          "pluginDetails": "\n----------- SCAN SUMMARY -----------\nKnown viruses: 4961442\nEngine version: 0.99.2\nScanned directories: 7\nScanned files: 6\nInfected files: 0\nData scanned: 0.41 MB\nData read: 0.41 MB (ratio 1.01:1)\nTime: 14.341 sec (0 m 14 s)\n",
          "htmlPluginDetails": false,
          "reports": []
        },
        {
          "id": "daa0133c-3149-45f3-ad1c-7acd921ab5e1-ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "jobId": "daa0133c-3149-45f3-ad1c-7acd921ab5e1",
          "sourceObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "sourceObjectClass": "org.roda.core.data.v2.ip.AIP",
          "sourceObjectOriginalId": "",
          "sourceObjectOriginalName": "",
          "outcomeObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "outcomeObjectClass": "org.roda.core.data.v2.ip.AIP",
          "outcomeObjectState": "INGEST_PROCESSING",
          "title": "AIP metadata validation",
          "dateCreated": 1476977345622,
          "dateUpdated": 1476977345956,
          "completionPercentage": 0,
          "stepsCompleted": 0,
          "totalSteps": 7,
          "plugin": "org.roda.core.plugins.plugins.base.DescriptiveMetadataValidationPlugin",
          "pluginName": "AIP metadata validation",
          "pluginVersion": "1.0",
          "pluginState": "SUCCESS",
          "pluginDetails": "",
          "htmlPluginDetails": false,
          "reports": []
        },
        {
          "id": "daa0133c-3149-45f3-ad1c-7acd921ab5e1-ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "jobId": "daa0133c-3149-45f3-ad1c-7acd921ab5e1",
          "sourceObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "sourceObjectClass": "org.roda.core.data.v2.ip.AIP",
          "sourceObjectOriginalId": "",
          "sourceObjectOriginalName": "",
          "outcomeObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "outcomeObjectClass": "org.roda.core.data.v2.ip.AIP",
          "outcomeObjectState": "INGEST_PROCESSING",
          "title": "AIP fixity information computation",
          "dateCreated": 1476977345974,
          "dateUpdated": 1476977346337,
          "completionPercentage": 0,
          "stepsCompleted": 0,
          "totalSteps": 7,
          "plugin": "org.roda.core.plugins.plugins.ingest.characterization.PremisSkeletonPlugin",
          "pluginName": "AIP fixity information computation",
          "pluginVersion": "1.0",
          "pluginState": "SUCCESS",
          "pluginDetails": "",
          "htmlPluginDetails": false,
          "reports": []
        },
        {
          "id": "daa0133c-3149-45f3-ad1c-7acd921ab5e1-ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "jobId": "daa0133c-3149-45f3-ad1c-7acd921ab5e1",
          "sourceObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "sourceObjectClass": "org.roda.core.data.v2.ip.AIP",
          "sourceObjectOriginalId": "",
          "sourceObjectOriginalName": "",
          "outcomeObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "outcomeObjectClass": "org.roda.core.data.v2.ip.AIP",
          "outcomeObjectState": "INGEST_PROCESSING",
          "title": "AIP file format identification (Siegfried)",
          "dateCreated": 1476977346418,
          "dateUpdated": 1476977346665,
          "completionPercentage": 0,
          "stepsCompleted": 0,
          "totalSteps": 7,
          "plugin": "org.roda.core.plugins.plugins.ingest.characterization.SiegfriedPlugin",
          "pluginName": "AIP file format identification (Siegfried)",
          "pluginVersion": "1.6.5",
          "pluginState": "SUCCESS",
          "pluginDetails": "",
          "htmlPluginDetails": false,
          "reports": []
        },
        {
          "id": "daa0133c-3149-45f3-ad1c-7acd921ab5e1-ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "jobId": "daa0133c-3149-45f3-ad1c-7acd921ab5e1",
          "sourceObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "sourceObjectClass": "org.roda.core.data.v2.ip.AIP",
          "sourceObjectOriginalId": "",
          "sourceObjectOriginalName": "",
          "outcomeObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "outcomeObjectClass": "org.roda.core.data.v2.ip.AIP",
          "outcomeObjectState": "INGEST_PROCESSING",
          "title": "Verify producer authorization",
          "dateCreated": 1476977346692,
          "dateUpdated": 1476977346759,
          "completionPercentage": 0,
          "stepsCompleted": 0,
          "totalSteps": 7,
          "plugin": "org.roda.core.plugins.plugins.ingest.VerifyProducerAuthorizationPlugin",
          "pluginName": "Verify producer authorization",
          "pluginVersion": "1.0",
          "pluginState": "SUCCESS",
          "pluginDetails": "Done with checking producer authorization for AIP ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "htmlPluginDetails": false,
          "reports": []
        },
        {
          "id": "daa0133c-3149-45f3-ad1c-7acd921ab5e1-ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "jobId": "daa0133c-3149-45f3-ad1c-7acd921ab5e1",
          "sourceObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "sourceObjectClass": "org.roda.core.data.v2.ip.AIP",
          "sourceObjectOriginalId": "",
          "sourceObjectOriginalName": "",
          "outcomeObjectId": "ab9bc35b-c9bb-4041-a98c-93fbf8d5b42c",
          "outcomeObjectClass": "org.roda.core.data.v2.ip.AIP",
          "outcomeObjectState": "ACTIVE",
          "title": "Auto accept",
          "dateCreated": 1476977346768,
          "dateUpdated": 1476977346886,
          "completionPercentage": 0,
          "stepsCompleted": 0,
          "totalSteps": 7,
          "plugin": "org.roda.core.plugins.plugins.ingest.AutoAcceptSIPPlugin",
          "pluginName": "Auto accept",
          "pluginVersion": "1.0",
          "pluginState": "SUCCESS",
          "pluginDetails": "",
          "htmlPluginDetails": false,
          "reports": []
        }
      ]
    }
  ],
  "facetResults": [],
  "date": 1476977803374
}
```

**Comment by rui-castro** [comment]
@luis100 solved in PR #649 

The issue that @jmaferreira mentioned was not the formating but the delimiter ','.
If we use delimiter ';' que csv will open in Excel correctly formated.

**Comment by jmaferreira** [comment]
Was the new delimiter defined for all possible exports?

Changing the delimiter from "," to ";" isn't just moving the problem from one place to another? CSV has a way os escaping column text. The CSV library can handle this (AFAIK).

**Comment by hsilva-keep** [comment]
@rui-castro the problem, at least if you try it in Libre Office, it's not related to the separator but instead to the new lines (as the problems start with the report column).

**Comment by rui-castro** [comment]
@jmaferreira @hsilva-keep the exported CSV is correctly formatted independently of the delimiter.

The thing with Excel not opening CSV correctly has to do with the delimiter that is **locale specific**. For european countries the delimiter is `;`, for others, like US, is `,`.

I've made an update to my PR #649 that makes the delimiter used configurable in roda-wui.properties.
There's a new configuration `csv.delimiter = ;`.

@hsilva-keep as for Libre Office, I've tried it and see no problem with new lines. See image below.

![screenshot from 2016-10-25 00 31 01](https://cloud.githubusercontent.com/assets/4218366/19668422/b6678e90-9a4f-11e6-9b4e-af1e6a1f2787.png)

**Comment by rui-castro** [comment]
@hsilva-keep anything missing in my PR #649 ?

**Comment by luis100** [comment]
@hsilva-keep please review and merge

**Comment by luis100** [comment]
Merged in lf_dev at eecb8ce60355b71b92172154630ae5e923129441
